### ISSUE TITLE
[clang-tidy] misc-redundant-expression warns on templated check for NaN

### ISSUE THREAD

**Issue by aeubanks** [llvm/llvm-project#92178]
clang-tidy properly doesn't warn on
```
bool isnan(float a) {
  return a != a;
}
```

however it does warn on the following:

```
$ cat /tmp/a.cc
template <typename T> bool isnan(T a) {
  return a != a;
}

bool isnan2(float a) {
  return isnan(a);
}

$ clang-tidy /tmp/a.cc --checks=misc-redundant-expression
1 warning generated.
/tmp/a.cc:2:12: warning: both sides of operator are equivalent [misc-redundant-expression]
    2 |   return a != a;
      |            ^
```

**Comment by vbvictor** [comment]
Maybe we should exclude templates
